config/config.go

```diff
@@ -119,6 +162,48 @@ func (c *MenderShellConfig) Validate() error {
 		}
 	}
 
+	//check if shell is given, if not, defaulting to /bin/sh
+	if c.ShellCommand == "" {
+		log.Warnf("ShellCommand is empty, defaulting to %s", DefaultShellCommand)
+		c.ShellCommand = DefaultShellCommand
+	}
+
+	if !filepath.IsAbs(c.ShellCommand) {
+		return errors.New("given shell (" + c.ShellCommand + ") is not an absolute path")
+	}
+
+	if !isExecutable(c.ShellCommand) {
+		return errors.New("given shell (" + c.ShellCommand + ") is not executable")
+	}
+
+	if c.User == "" {
+		return errors.New("please provide a user to run the shell as")
+	}
+	u, err := user.Lookup(c.User)
+	if err == nil && u == nil {
+		return errors.New("unknown error while getting a user id")
+	}
+	if err != nil {
+		return err
+	}
+
+	if !isInShells(c.ShellCommand) {
+		log.Warnf("ShellCommand %s is not present in /etc/shells", DefaultShellCommand)
+	}
+
+	if c.Terminal.Width != 0 {
+		c.Terminal.Width = DefaultTerminalWidth
+	}
+
+	if c.Terminal.Height != 0 {
+		c.Terminal.Height = DefaultTerminalHeight
+	}
```

Comment:
This looks incorrect.

```suggestion
	if c.Terminal.Width == 0 {
		c.Terminal.Width = DefaultTerminalWidth
	}

	if c.Terminal.Height == 0 {
		c.Terminal.Height = DefaultTerminalHeight
	}
```

Reply:
right. thanks.